keps/sig-apps/961-maxunavailable-for-statefulset/README.md

```diff
@@ -263,17 +458,110 @@ When downgrading from a release with this feature, to a release without maxUnava
+
+Below are some examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Gather feedback from developers and surveys
+- Complete features A, B, C
+- Additional tests are in Testgrid and linked in KEP
+
+#### GA
+
+- N examples of real-world usage
+- N installs
+- More rigorous forms of testing—e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+-->
+
 - Alpha: Initial support for maxUnavailable in StatefulSets added. Disabled by default with default value of 1.
-- Beta:  Enabled by default with default value of 1 with upgrade downgrade testedd at least manually.
+- Beta:  Enabled by default with default value of 1 with upgrade downgrade tested at least manually.
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+We will default to 1 for maxUnavailable field in StatefulSet for backward compatibility
+
+Downgrades
+
+When downgrading from a release with this feature, to a release without maxUnavailable, there are two cases
+- If maxUnavailable is greater than 1, there are two more cases:-
+  - If you're rolling back to a release that doesn't have this field - then there is even no way to discover it
+  - If you're just disabling the feature (either together with downgrade to a release that has a field or without downgrade),the field should remain set
+       (unless someone will explicitly delete it later), but controller should ignore its behavior (and there shouldn't be a way to set it if the feature gate
+       is switched off).
+ - If maxUnavailable is less than equal to 1 -- in this case user wont see any difference in behavior -->
+
+### Version Skew Strategy
 
+No.
```

Comment:
You're coordinating between kube-apiserver and kube-controller-manager the enablement of this feature. Please describe what happens when each of these components has the feature on and off accordingly. I'd expect a matrix with 4 entries for each of the components  having the feature on and off.

Reply:
Make sense

Reply:
Make sense